packages/flutter_tools/lib/src/run_hot.dart

```diff
@@ -262,14 +263,27 @@ class HotRunner extends ResidentRunner {
 
     firstBuildTime = DateTime.now();
 
+    final List<Future<bool>> startupTasks = <Future<bool>>[];
     for (FlutterDevice device in flutterDevices) {
-      final int result = await device.runHot(
+      if (device.generator != null) {
+        startupTasks.add(
```

Comment:
As discussed offline, some comments here about why this is safe, and at the other callsites about why they are still there would be helpful.

Reply:
Done